### ISSUE TITLE
[BUG] gc-cons-threshold at 2^61

### ISSUE THREAD

**Issue by jsravn** [hlissner/doom-emacs#2103]
**Describe the issue**
I noticed `gc-cons-threshold` is 2^61 after doom finishes starting up. I think it's supposed to be restored to 16mib or some other value.

**Steps to reproduce**
1. `M-:` then `gc-cons-threshold` shows it set to 2^61.

**System information**
<details>

```
emacs   version    26.3
        features   XPM JPEG TIFF GIF PNG RSVG IMAGEMAGICK SOUND GPM DBUS GSETTINGS GLIB NOTIFY ACL GNUTLS LIBXML2 FREETYPE M17N_FLT LIBOTF XFT ZLIB TOOLKIT_SCROLL_BARS GTK3 X11 XDBE XIM MODULES THREADS LIBSYSTEMD LCMS2
        build      Aug 29, 2019
        buildopts  (--prefix=/usr --sysconfdir=/etc --libexecdir=/usr/lib --localstatedir=/var --with-x-toolkit=gtk3 --with-xft --with-modules 'CFLAGS=-march=x86-64 -mtune=generic -O2 -pipe -fno-plt' CPPFLAGS=-D_FORTIFY_SOURCE=2 LDFLAGS=-Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now)
        windowsys  x
        daemonp    server-running
doom    version    2.0.9
        build      HEAD -> develop 2dc50f49f 2019-11-22 00:39:55 -0500
system  type       gnu/linux
        config     x86_64-pc-linux-gnu
        shell      /bin/zsh
        uname      Linux 5.3.11-arch1-1 #1 SMP PREEMPT Tue, 12 Nov 2019 22:19:48 +0000 x86_64
        path       (~/.cargo/bin ~/.nvm/versions/node/v9.4.0/bin ~/.pyenv/shims ~/.jabba/jdk/adopt@1.8.0-232/bin ~/.pyenv/bin ~/.cargo/bin ~/.cabal/bin ~/.local/bin ~/go/bin ~/godev/bin ~/.emacs.d/bin /opt/google-cloud-sdk/bin /usr/local/bin /usr/local/sbin /usr/bin /var/lib/flatpak/exports/bin /usr/lib/jvm/default/bin /usr/bin/site_perl /usr/bin/vendor_perl /usr/bin/core_perl /usr/lib/emacs/26.3/x86_64-pc-linux-gnu)
config  envfile    nil
        elc-files  0
        modules    (:completion company helm :ui doom doom-dashboard doom-quit fill-column hl-todo modeline nav-flash ophints (popup +all +defaults) (pretty-code +iosevka) treemacs vc-gutter vi-tilde-fringe window-select workspaces :editor (evil +everywhere) file-templates fold format multiple-cursors rotate-text snippets :emacs dired electric vc :term term :tools docker editorconfig eval flycheck (lookup +docsets) lsp magit make terraform :lang data emacs-lisp (go +lsp) javascript markdown (org +dragndrop +ipython +pandoc +present) (python +lsp +pyenv) (scala +lsp) sh :config (default +bindings +smartparens))
        packages   (n/a)
        elpa       (n/a)
```
</details>

**Comment by jsravn** [comment]
Okay, I see this was changed recently https://github.com/hlissner/doom-emacs/commit/db19241936e56bb0f406f7c3a3602908967e4f77. Well after upgrade my emacs is stuttering/hanging a lot - so I suspect that may be the issue. I'll try disabling it and see what happens.

**Comment by jsravn** [comment]
Reverting that commit fixes all the hanging I was getting. Maybe it's an issue w/ lsp-mode since that's where I was suffering.

**Comment by Emantor** [comment]
Did you `bin/doom update`to install gcmh?

**Comment by ckruse** [comment]
Oh, nice! Sadly here it is installed but not loaded. When I manually start `gcmh-mode` it loads (and seems to be active)

**Comment by ckruse** [comment]
It doesn't get loaded because `doom-interactive-mode` is `t`, I guess:

~~~elisp
(unless doom-interactive-mode
  (add-transient-hook! 'pre-command-hook (gcmh-mode +1))
  (with-eval-after-load 'gcmh
    (setq gcmh-idle-delay 10
          gcmh-verbose doom-debug-mode)
    (add-hook 'focus-out-hook #'gcmh-idle-garbage-collect)))
~~~

Is this a logic error? Should it be `(when)` instead of `(unless)`?

**Comment by jsravn** [comment]
> Did you bin/doom updateto install gcmh?

Yes.

**Comment by hlissner** [comment]
> Is this a logic error? Should it be (when) instead of (unless)?

Yes, it was.

---

This issue should be resolved in 34500c9. Let me know if that isn't the case and I'll reopen it. Thanks for bringing it to my attention!

**Comment by ckruse** [comment]
Thank you, @hlissner, for your tireless work on this project!
